sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuCompressedColumnVector.java

```diff
@@ -109,6 +111,44 @@ public static ColumnarBatch from(DeviceMemoryBuffer compressedBuffer, TableMeta
     return new ColumnarBatch(columns, (int) rows);
   }
 
+
+  /**
+   * Build a columnar batch from a compressed data buffer and specified table metadata
+   * NOTE: The data remains compressed and cannot be accessed directly from the columnar batch.
+   */
+  public static ColumnarBatch from(DeviceMemoryBuffer compressedBuffer,
+      TableMeta tableMeta,
+      DataType[] colTypes) {
+    long rows = tableMeta.rowCount();
+    if (rows != (int) rows) {
+      throw new IllegalStateException("Cannot support a batch larger that MAX INT rows");
+    }
+
+    ColumnMeta columnMeta = new ColumnMeta();
+    int numColumns = tableMeta.columnMetasLength();
+    assert numColumns == colTypes.length : "Size mismatch on types";
+    ColumnVector[] columns = new ColumnVector[numColumns];
+    try {
+      for (int i = 0; i < numColumns; ++i) {
+        tableMeta.columnMetas(columnMeta, i);
+        DataType type = colTypes[i];
+        // TODO we need a good way to verify that we can map between the DType(s) we have in
+        //  columnMeta and the types passed in.
```

Comment:
Can't we call `GpuColumnVector.getRapidsType` and verify it matches the DType in the `ColumnMeta`?

Reply:
That does not work for nested types. I could do some checks and do it for non-nested types, but I will write something to make this work properly.